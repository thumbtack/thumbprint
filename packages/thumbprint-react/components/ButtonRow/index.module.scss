@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';
@import '~@thumbtack/thumbprint-scss/mixins';

.root {
    display: flex;
    align-items: center;
}

.item + .item {
    margin-left: $tp-space__2;
}

.justifyCenter {
    justify-content: center;
}

.justifyLeft {
    justify-content: flex-start;
}

.justifyRight {
    justify-content: flex-end;
}

.stackBelowSmall {
    @include tp-respond-below($tp-breakpoint__small) {
        flex-direction: column-reverse;
        align-items: stretch;

        .item + .item {
            margin-bottom: $tp-space__2;
            margin-left: 0;
        }
    }
}
