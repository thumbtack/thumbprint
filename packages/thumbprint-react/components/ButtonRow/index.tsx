import React from 'react';
import classNames from 'classnames';
import styles from './index.module.scss';

interface PropTypes {
    /**
     * `Button` components to render.
     */
    children?: React.ReactNode;
    /**
     * Controls the horizontal alignment of buttons within the container.
     */
    justify?: 'center' | 'left' | 'right';
    /**
     * Stack items below the small breakpoint. This pairs well with `width="full-below-small"`
     * for buttons
     */
    isStackedBelowSmall?: boolean;
    /**
     * A selector hook into the React component for use in automated testing environments.
     */
    dataTest?: string;
}

export default function ButtonRow({
    children = null,
    justify = 'left',
    dataTest,
    isStackedBelowSmall = false,
}: PropTypes): JSX.Element {
    return (
        <div
            className={classNames({
                [styles.root]: true,
                [styles.stackBelowSmall]: isStackedBelowSmall,
                [styles.justifyCenter]: justify === 'center',
                [styles.justifyLeft]: justify === 'left',
                [styles.justifyRight]: justify === 'right',
            })}
            data-test={dataTest}
        >
            {React.Children.map(children, child => (
                <div className={styles.item}>{child}</div>
            ))}
        </div>
    );
}
