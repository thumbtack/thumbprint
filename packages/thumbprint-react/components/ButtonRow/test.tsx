import React from 'react';
import { mount } from 'enzyme';
import ButtonRow from './index';

test('renders content that is passed in', () => {
    const wrapper = mount(<ButtonRow>goose</ButtonRow>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

describe('placement', () => {
    test('default (left)', () => {
        const wrapper = mount(<ButtonRow justify="left" />);
        expect(wrapper).toMatchSnapshot();
    });

    test('center', () => {
        const wrapper = mount(<ButtonRow justify="center" />);
        expect(wrapper).toMatchSnapshot();
    });

    test('right', () => {
        const wrapper = mount(<ButtonRow justify="right" />);
        expect(wrapper).toMatchSnapshot();
    });
});

test('adds `dataTest` prop', () => {
    const wrapper = mount(<ButtonRow dataTest="Duck" />);
    expect(wrapper.find('div').prop('data-test')).toEqual('Duck');
    expect(wrapper).toMatchSnapshot();
});
