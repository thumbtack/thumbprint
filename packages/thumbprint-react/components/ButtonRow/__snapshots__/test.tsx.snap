// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds \`dataTest\` prop 1`] = `
<ButtonRow
  dataTest="Duck"
>
  <div
    className="root justifyLeft"
    data-test="Duck"
  />
</ButtonRow>
`;

exports[`placement center 1`] = `
<ButtonRow
  justify="center"
>
  <div
    className="root justifyCenter"
  />
</ButtonRow>
`;

exports[`placement default (left) 1`] = `
<ButtonRow
  justify="left"
>
  <div
    className="root justifyLeft"
  />
</ButtonRow>
`;

exports[`placement right 1`] = `
<ButtonRow
  justify="right"
>
  <div
    className="root justifyRight"
  />
</ButtonRow>
`;

exports[`renders content that is passed in 1`] = `
<ButtonRow>
  <div
    className="root justifyLeft"
  >
    <div
      className="item"
      key=".0"
    >
      goose
    </div>
  </div>
</ButtonRow>
`;
